sdks/python/apache_beam/io/hdfilesystem.py

```diff
@@ -0,0 +1,190 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""FileSystem implementation for accessing Hadoop Distributed File System
+files."""
+
+from __future__ import absolute_import
+
+import posixpath
+
+from hdfs3 import HDFileSystem as HDFS3
+
+from apache_beam.io.filesystem import BeamIOError
+from apache_beam.io.filesystem import CompressedFile
+from apache_beam.io.filesystem import CompressionTypes
+from apache_beam.io.filesystem import FileMetadata
+from apache_beam.io.filesystem import FileSystem
+from apache_beam.io.filesystem import MatchResult
+
+__all__ = ['HdFileSystem']
+
+
+COPY_BUFFER_SIZE = 2**16
+
+
+# TODO(udim): Add @retry.with_exponential_backoff to some functions, like in
+# gcsio.py.
+
+
+class HdFileSystem(FileSystem):
+  """FileSystem implementation that supports HDFS."""
+
+  def __init__(self, *args, **kwargs):
+    """Initialize a connection to HDFS.
+
+    Keyword arguments:
+      host: Hostname of HDFS server to connect to.
+      port: Port of HDFS server to connect to.
+    """
+    host = kwargs.pop('host')
```

Comment:
<!--thread_id:cc_156243873_t; commit:3c708ac10af4457fe46e07503f0b72930832db07; resolved:1-->
<!--section:context-quote-->
> **chamikaramj** wrote:
> We might not be able to do this in all cases (we might have to get this values from the environment). Check for keys, host in kwargs and raise a NotImplementedError ?

<!--section:body-->
Now uses defaults and config file found using common environment variables.